server/admin/admin_server_test.go

```diff
@@ -318,10 +347,11 @@ func TestServer_CreateTree(t *testing.T) {
 	keySignatureMismatch.SignatureAlgorithm = sigpb.DigitallySigned_RSA
 
 	tests := []struct {
-		desc                           string
-		req                            *trillian.CreateTreeRequest
-		createErr                      error
-		commitErr, wantErr, wantCommit bool
+		desc                  string
+		req                   *trillian.CreateTreeRequest
+		createErr             error
+		commitErr, wantCommit bool
+		wantErr               string
```

Comment:
I agree, and never usually check the error string but, in this case, there are a lot of different errors that could occur. I really want to know that the test is failing for the reason I think it is, rather than because the test is misconfigured, for example. Error codes would be a better solution except, as you suggest, I don't think they'd actually be different for most of these test cases (they're almost all cases of invalid arguments). Maybe that would be good enough for distinguishing the intended type of error from unintended errors though?